### PR TITLE
Add pkg for System.Diagnostics.PerformanceCounter

### PR DESCRIPTION
**Pull request by adiaaida**
This change adds System.Diagnostics.PerformanceCounter to the compat
pack.

### CODE REVIEW

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -1364,6 +1364,9 @@
         "xamarinmac20": "Any",
         "xamarintvos10": "Any",
         "xamarinwatchos10": "Any"
+      },
+      "AssemblyVersionInPackageVersion": {
```
Comment (danmosemsft) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Does this need 
```
      "InboxOn": {},
```
?
Like in 4e408ff42f9ecbd63f6787e781d6f06bb28ca81e

File: pkg/descriptions.json [hunk 2]
```diff
@@ -566,6 +566,13 @@
     ]
   },
   {
+    "Name": "System.Diagnostics.PerformanceCounter",
```
Comment (danmosemsft) [inline pkg/descriptions.json]: Given you have this, I don't understand why you're getting `
D:\j\workspace\windows-TGrou---0d2c9ac4\Tools\Packaging.targets(895,5): error : Unable to find description for package System.Diagnostics.PerformanceCounters [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Diagnostics.PerformanceCounter\pkg\System.Diagnostics.PerformanceCounters.pkgproj]`

@weshaggard ?

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 1 above)
Comment (adiaaida) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: There's already an InboxOn line in the PerformanceCounter section, so I thought it wasn't necessary.

File: src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounters.pkgproj [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (weshaggard) [inline src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounters.pkgproj]: You should rename this pkgproj to System.Diagnostics.PerformanceCounter without the "s". That will fix your cannot find description error.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 1 above)
Comment (adiaaida) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Ok, now I'm getting errors saying that we shouldn't be supporting PerformanceCounter on xamarin, etc, so I'll remove those lines in the InboxOn section.

File: src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounters.pkgproj (hunk 3 above)
Comment (danmosemsft) [reply src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounters.pkgproj]: aha!

**Comment by danmosemsft** [comment]
´D:\j\workspace\windows-TGrou---0d2c9ac4\Tools\Packaging.targets(1106,5): error : System.Diagnostics.PerformanceCounter should not be supported on MonoAndroid,Version=v1.0 but has both compile and runtime assets. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Diagnostics.PerformanceCounter\pkg\System.Diagnostics.PerformanceCounter.pkgproj]´

**Comment by danmosemsft** [comment]
@weshaggard ?

**Comment by adiaaida** [comment]
@weshaggard do you know what's going on with the Mono/Xamarin build failures here? Keeping them in the InboxOn section causes the packaging build to fail, but removing them does too (and doesn't seem like the right move, since they were already there).

**Comment by weshaggard** [comment]
>@weshaggard do you know what's going on with the Mono/Xamarin build failures here? Keeping them in the InboxOn section causes the packaging build to fail, but removing them does too (and doesn't seem like the right move, since they were already there).

Add `$(AllXamarinFrameworks)` to the list of supported platforms like https://github.com/dotnet/corefx/blob/905e3716483755a5568ba918198e195bc5627083/src/System.ServiceProcess.ServiceController/pkg/System.ServiceProcess.ServiceController.pkgproj#L6.

**Comment by danmosemsft** [comment]
@dotnet/dnceng please take a look:
```
16:23:26   Restoring packages for /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/init-tools.msbuild...
16:23:26   Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.dotnet.build.tasks.feed/index.json'.
16:23:26   Access to the path '/tmp/NuGet/TempCache/6fe417a8-aa75-46ed-9e46-2305161dafb0' is denied.
16:23:26     Permission denied
16:23:26   Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.dotnet.buildtools/index.json'.
16:23:26   Access to the path '/tmp/NuGet/TempCache/6fe417a8-aa75-46ed-9e46-2305161dafb0' is denied.
16:23:26     Permission denied
16:23:26   Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.dotnet.build.tasks.feed/index.json'.
16:23:26   Access to the path '/tmp/NuGet/TempCache/6fe417a8-aa75-46ed-9e46-2305161dafb0' is denied.
16:23:26     Permission denied
16:23:26   Retrying 'FindPackagesByIdAsync' for source 'https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-buildtools/nuget/v3/flatcontainer/microsoft.dotnet.buildtools/index.json'.
16:23:26   Access to the path '/tmp/NuGet/TempCache/6fe417a8-aa75-46ed-9e46-2305161dafb0' is denied.
16:23:26     Permission denied
```

**Comment by danmosemsft** [comment]
Now various errors like
```

D:\j\workspace\windows-TGrou---0d2c9ac4\Tools\Packaging.targets(1106,5): error : Framework MonoAndroid,Version=v1.0 should support System.Diagnostics.PerformanceCounter inbox but contained a reference assemblies: ref/netstandard2.0/System.Diagnostics.PerformanceCounter.dll.  You may need to add <InboxOnTargetFramework Include="monoandroid10" /> to your project. [D:\j\workspace\windows-TGrou---0d2c9ac4\src\System.Diagnostics.PerformanceCounter\pkg\System.Diagnostics.PerformanceCounter.pkgproj]
```
I need to understand this, meanwhile pushed a change to do this.

**Comment by danmosemsft** [comment]
@dotnet-bot test OSX x64 Debug Build please

**Comment by danmosemsft** [comment]
@adiaaida well I can't push to your branch. maybe add me as a collaborator on your fork? meanwhile you can add this yourself to the pkgproj:
```xml
   </ItemGroup>
+  <ItemGroup>
+    <InboxOnTargetFramework Include="MonoAndroid10" />
+    <InboxOnTargetFramework Include="MonoTouch10" />
+    <InboxOnTargetFramework Include="xamarinios10" />
+    <InboxOnTargetFramework Include="xamarinmac20" />
+    <InboxOnTargetFramework Include="xamarintvos10" />
+    <InboxOnTargetFramework Include="xamarinwatchos10" />
+  </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
```

**Comment by adiaaida** [comment]
@danmosemsft Thank you. Looks like the packaging leg is green now.

File: src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounter.pkgproj [hunk 4]
```diff
@@ -7,5 +7,13 @@
     </ProjectReference>
     <ProjectReference Include="..\src\System.Diagnostics.PerformanceCounter.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <InboxOnTargetFramework Include="MonoAndroid10" />
```
Comment (weshaggard) [inline src/System.Diagnostics.PerformanceCounter/pkg/System.Diagnostics.PerformanceCounter.pkgproj]: I suggest doing:

```
 <InboxOnTargetFramework Include="$(AllXamarinFrameworks)" />
```

File: src/System.Diagnostics.PerformanceCounter/src/Configurations.props [hunk 5]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <BuildConfigurations>
```
Comment (weshaggard) [inline src/System.Diagnostics.PerformanceCounter/src/Configurations.props]: @danmosemsft we are still going to need configurations for netcoreapp, see https://github.com/dotnet/corefx/issues/24776, do you want to do those all in one pass or as folks are working on the library?

Comment (danmosemsft) [reply src/System.Diagnostics.PerformanceCounter/src/Configurations.props]: I guess we should try to catch this in PR's then do a pass for all the rest.

Comment (danmosemsft) [reply src/System.Diagnostics.PerformanceCounter/src/Configurations.props]: I think it's fine for @adiaaida to submit this without and @safern will fix it in his pass.

Comment (safern) [reply src/System.Diagnostics.PerformanceCounter/src/Configurations.props]: We will still need to do a pass to make sure everything is marked as inbox correctly for Xamarin frameworks once we talk to @marek-safar next week that he'll be around. So it's ok if I take that as part of that pass :)

**Comment by weshaggard** [comment]
test Windows x64 Debug Build

File: src/System.Diagnostics.PerformanceCounter/src/Configurations.props (hunk 5 above)
Comment (weshaggard) [reply src/System.Diagnostics.PerformanceCounter/src/Configurations.props]: OK I'm fine with doing it in one pass.

**Comment by danmosemsft** [comment]
OK then I will merge this and @safern can follow up as above.

**Pull request merged**